app/code/community/Metrilo/Analytics/Helper/Data.php

```diff
@@ -265,7 +265,7 @@ private function _buildOrdersForSubmition($orders) {
         $ordersForSubmition = array();
 
         foreach ($orders as $order) {
-            if ($order->getId() && $order->getStatus() != null) {
+            if ($order->getId() && $order->getStatus() != null && $order->getCustomerEmail()) {
```

Comment:
What if the email is whitespace?

Reply:
Usually checkout fields have js validators assigned.

Reply:
Be prepared to get some kicks from the new tracking with field validations 🗡

Reply:
is `'' == true` in PHP?

Reply:
is there anything that can strip whitespace and then check presence?

Reply:
trim() method is removing white spaces. As evgeni mentioned there is going to be troubles with validation, but that should come before the data is inserted in the database. Adding this check 'trim($order->getCustomerEmail())' will ensure there are no errors when excluding empty mail orders. I assumed it's not necessary but will add it in the following commit.